### DISCUSSION TITLE
Hardware decoding seems causing a frame-back-step bug.

### DISCUSSION THREAD

**Discussion by 1000Fear** [mpv-player/mpv#15435] [Q&A]
When I use hardware decoding and simultaneously press frame-back-step to search backward, in some videos, there is a loop between several frames.

For example, here is a sequence of frames: `1-2-3-4-5`. If I start from frame 4 and search backward to frame 3 and then frame 2, when I reach frame 2, it doesn't proceed to frame 1 but instead jumps back to frame 4. This creates a dead loop within this range. When I hold down the frame-back-step key, regardless of whether I start from frame 5 or 4, it ends up repeating `5-4-3-2-4-3-2-...` instead of `5-4-3-2-1`.

This issue only occurs with a few videos; most video files work fine when searching backward, or it doesn't happen when using software decoding. What could be causing this? Is it related to mpv, or is it an issue with the video file?

hardware decode config (d3d11):

> hwdec=d3d11va
gpu-api=d3d11
vo=gpu-next
gpu-context=d3d11

hardware decode config (vulkan):

> hwdec=vulkan
gpu-api=vulkan
vo=gpu-next
gpu-context=winvk

Other hardware decoding options on Windows platforms cause the same issue; only using software decoding prevents the bug from occurring.

**Comment by llyyr** [comment]
frame backstepping is a hack and in general can break depending on what demuxer/decoder was used, or whether any post filtering was used. There isn't much that mpv can do here.
